edi_storage_oca/models/edi_backend.py

```diff
@@ -0,0 +1,173 @@
+# Copyright 2020 ACSONE SA
+# @author Simone Orsi <simahawk@gmail.com>
+# Copyright 2021 ForgeFlow S.L. (https://www.forgeflow.com)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+import logging
+import os
+
+from odoo import fields, models
+
+from .. import utils
+
+_logger = logging.getLogger(__name__)
+
+
+class EDIBackend(models.Model):
+    _inherit = "edi.backend"
+
+    storage_id = fields.Many2one(
+        string="FS Storage",
+        comodel_name="fs.storage",
+        help="Storage for in-out files",
+        ondelete="restrict",
+    )
+    """
+    We assume the exchanges happen it 2 ways (input, output)
+    and we have a hierarchy of directory like:
+
+        from_A_to_B
+            |- pending
+            |- done
+            |- error
+        from_B_to_A
+            |- pending
+            |- done
+            |- error
+
+    where A and B are the partners exchanging data and they are in turn
+    sender and receiver and vice versa.
+    """
+    # TODO: these paths should probably be by type instead
+    # Here we can maybe set a common root folder for this exchange.
+    input_dir_pending = fields.Char(
+        "Input pending directory", help="Path to folder for pending operations"
+    )
+    input_dir_done = fields.Char(
+        "Input done directory", help="Path to folder for doneful operations"
+    )
+    input_dir_error = fields.Char(
+        "Input error directory", help="Path to folder for error operations"
+    )
+    output_dir_pending = fields.Char(
+        "Output pending directory", help="Path to folder for pending operations"
+    )
+    output_dir_done = fields.Char(
+        "Output done directory", help="Path to folder for doneful operations"
+    )
+    output_dir_error = fields.Char(
+        "Output error directory", help="Path to folder for error operations"
+    )
+
+    _storage_actions = ("check", "send", "receive")
```

Comment:
this parts can be dropped if we don't use components anymore

Reply:
Sure